pagefaultgames/pokerogue
# [Misc] Add battle music for Legendary Pokémon forms

## What are the changes the user will see?

Adds more battle music for Legendary Pokémon with alternate forms, specifically Dialga, Palkia, and Calyrex. Dialga and Palkia will get their Legends Arceus Origin Form battle theme, and Calyrex will get its Ice & Shadow Rider battle theme.

## Why am I making these changes?



These specific Pokémon have battle themes for their alternate forms, and can be utilized if they are either encountered in the wild or as a static boss encounter, like what I mentioned in #3054 .

## What are the changes from a developer perspective?


Enemy Dialga, Palkia, and Calyrex now play different music based on their form key, with the existing music playing for their base forms, and the new tracks playing for their alternate forms.

### Screenshots/Videos



https://github.com/user-attachments/assets/7c64ce5c-9c69-467c-8a0f-d19d1fdf2b04

## How to test the changes?




There isn't a way to force the form index of an opponent at present, so the easiest way to test would be to temporarily add Dialga, Palkia, or Calyrex to the `getSpeciesFormIndex` in `battle-scene.ts` and encounter the forms randomly to see the music changes.

## Checklist
